src/librustc/ich/impls_hir.rs

```diff
@@ -349,14 +349,22 @@ impl_stable_hash_for!(struct hir::FnDecl {
     inputs,
     output,
     variadic,
-    has_implicit_self
+    implicit_self
```

Comment:
pre-existing, but it's kind of weird that we call this an "implicit" self, given that we're talking about an explicitly written `self` annotation

Reply:
I think it makes sense if you consider the "implicit"-ness of it that you don't specify the type, whereas in the explicit case you do.